pkgs/tools/networking/narrowlink/default.nix

```diff
@@ -0,0 +1,46 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, openssl
+, stdenv
+, darwin
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "narrowlink";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "narrowlink";
+    repo = "narrowlink";
+    rev = version;
+    hash = "sha256-Zq2tzO5g1+Hd+Kgep5pAxf/hQDFCSzEbOqnktxsifnk=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
+
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
```

Comment:
Would be nice if upstream kept this in version control.  Maybe ask them?

Reply:
Done - https://github.com/narrowlink/narrowlink/commit/37def5c5cfd62a4d623d92fdf03d4e8d3eac0e08

Reply:
@SajjadPourali can you please cut a release after the lock-file addition commit? That would make automating updating this package a bit simpler here. 

Also since you commented, do you have any idea why the build is failing on `x86_64-darwin` - build logs here - https://logs.ofborg.org/?key=nixos/nixpkgs.248723&attempt_id=d7ae30ff-d5b7-4cb0-9d55-884033c82973

Reply:
I think the failure is due to using Apple SDK 10 instead of 11. Please take a look at this issue for more info: https://github.com/GuillaumeGomez/sysinfo/issues/915

------

For the tag/release, does https://github.com/narrowlink/narrowlink/releases/tag/Nightly works for you? It might take a bit time for the actual release.

Reply:
Thanks! The build failure is solved! 

Using the `Nightly` tag will imply making the version `unstable-DDDD-MM-YY`. Do you think cutting a minor or patch release is possible?

There are workarounds

Reply:
The new release will be done after https://github.com/narrowlink/narrowlink/pull/28. I plan to finish it by this weekend.

Reply:
> Thanks! The build failure is solved!
> 
> Using the `Nightly` tag will imply making the version `unstable-DDDD-MM-YY`. Do you think cutting a minor or patch release is possible?
> 
> There are workarounds but not clean.

@dit7ya Done

Reply:
Done, the Cargo.lock is still outdated but I figured using a patch (much smaller than checking in the entire lockfile) is faster than waiting for the next release.